plugin/fuzz/source/dextool/plugin/fuzz/backend/fuzzvariant.d

```diff
@@ -0,0 +1,628 @@
+        FileName gmock;
+        FileName pre_incl;
+        FileName post_incl;
+    }
+
+    Files getFiles();
+    MainNs getMainNs();
+    MainInterface getMainInterface();
+    StubPrefix getArtifactPrefix();
+    DextoolVersion getToolVersion();
+    CustomHeader getCustomHeader();
+
+    SUTEnvironment getSut();
+}
+
+/// Data produced by the generator like files.
+@safe interface Products {
+    void putFile(FileName fname, CppHModule hdr_data);
+    void putFile(FileName fname, CppModule impl_data);
+    void putLocation(FileName loc, LocationType type);
+}
+
+/// Generator of test doubles for C++ code.
+struct Generator {
+    import cpptooling.data.representation : CppRoot;
+    import cpptooling.data.symbol.container : Container;
+
+    private static struct Modules {
+        import dextool.plugin.utility : MakerInitializingClassMembers;
+
+        mixin MakerInitializingClassMembers!Modules;
+
+        CppModule hdr;
+        CppModule impl;
+        CppModule gmock;
+    }
+
+     void process(ref CppRoot root, ref Container container) {
+        import cpptooling.data.symbol.types : USRType;
+
+        //auto fl = rawFilter(root, products, (USRType usr) => container.find!LocationTag(usr));
+        //logger.trace("Filtered:\n", fl.toString());
+
+        //auto impl_data = translate(fl, container, params);
+        //logger.trace("Translated to implementation:\n", impl_data.toString());
+        logger.trace("kind:\n", impl_data.kind);
+
+        auto modules = Modules.make();
+        generate(impl_data, params, modules, container);
+        postProcess(params, products, modules);
+    }
+
+private:
+    Parameters params;
+    Products products;
+    
+    static void postProcess(Parameters params, Products prods, Modules modules) {
+        import cpptooling.generator.includes : convToIncludeGuard,
+            generatetPreInclude, generatePostInclude, makeHeader;
+
+        static auto outputHdr(CppModule hdr, FileName fname, DextoolVersion ver,
+                CustomHeader custom_hdr) {
+            auto o = CppHModule(convToIncludeGuard(fname));
+            o.header.append(makeHeader(fname, ver, custom_hdr));
+            o.content.append(hdr);
+
+            return o;
+        }
+
+        static auto output(CppModule code, FileName incl_fname, FileName dest,
+                DextoolVersion ver, CustomHeader custom_hdr) {
+            import std.path : baseName;
+
+            auto o = new CppModule;
+            o.suppressIndent(1);
+            o.append(makeHeader(dest, ver, custom_hdr));
+            o.include(incl_fname.baseName);
+            o.sep(2);
+            o.append(code);
+
+            return o;
+        }
+
+	prods.putFile(params.getFiles.hdr, outputHdr(modules.hdr, params.getFiles.hdr, params.getToolVersion, params.getCustomHeader));
+        prods.putFile(params.getFiles.impl, output(modules.impl, params.getFiles.hdr, params.getFiles.impl, params.getToolVersion, params.getCustomHeader));
```

Comment:
```bash
dub fetch dfmt --version0.5.0-beta4
cd dextool
dub run dfmt -- <some file>
```

I think the above uses the .editorconfig in the root of dextool. otherwise supply it with -c.